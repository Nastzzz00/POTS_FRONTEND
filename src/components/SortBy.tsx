import React, { Fragment } from 'react'
import { Typography, Select } from 'antd'

const { Text } = Typography
const { Option } = Select

export interface ISortyBy {
  state?: any
  setState?: any
  sorts?: any
  title?: any
  sortby?: any
}

const SortBy = (props: ISortyBy) => {
  const { state, setState, sorts, sortby } = props
  const handleChange = (value: any, state: any, setState: any) => {
    setState({
      ...state,
      sortby: value,
    })
  }
  return (
    <Fragment>
      <div style={{ marginRight: '10px' }}>
        <Text>Sort By: </Text>
      </div>
      <div>
        <Select
          defaultValue={'status'}
          value={state.sortby}
          style={{ width: 200 }}
          onChange={(value: any) => {
            handleChange(value, state, setState)
          }}>
          {sorts.map((items: any) => (
            <Option value={items.value}>{items.desc}</Option>
          ))}
        </Select>
      </div>
    </Fragment>
  )
}

export default SortBy
