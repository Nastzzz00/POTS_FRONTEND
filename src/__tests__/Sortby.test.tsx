/**
 * @format
 */

import React from 'react'
import { shallow } from 'enzyme'
// import ShallowRenderer from "react-test-renderer/shallow";

// // Note: test renderer must be required after react-native.
// import renderer from "react-test-renderer";

import SortBy from '../components/SortBy'

const sorts = [{ value: 'Something', desc: 'Something again' }]
const state = { sortby: 'Some value' }
const props = {
  title: 'SortBy',
  sorts: sorts,
}

describe('SortBy', () => {
  test('renders correctly', () => {
    expect(
      shallow(
        <SortBy
          {...props}
          state={state}
          sorts={sorts}
          setState={{ ...state, sortby: 'value' }}
        />,
      ),
    ).toMatchSnapshot()
  })
})
