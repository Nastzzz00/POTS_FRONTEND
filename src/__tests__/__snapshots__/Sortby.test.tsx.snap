// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortBy renders correctly 1`] = `ShallowWrapper {}`;
